shared/reducers/fs.js

```diff
@@ -10,7 +10,22 @@ export default function(state: Types.State = initialState, action: FSGen.Actions
     case FSGen.resetStore:
       return initialState
     case FSGen.folderListLoaded:
-      return state.mergeIn(['pathItems'], action.payload.pathItems)
+      const toMerge = action.payload.pathItems.map((item, path) => {
+        if (item.type !== 'folder') {
+          return item
+        }
+        const original = state.pathItems.get(path)
+        if (original && original.progress === 'loaded' && item.progress === 'pending') {
```

Comment:
unclear that this is actually what you want to do.
1. your saga likely wants to just avoid doing this work so you could see if you already have the original and just bail on doing any loading
2. this means if you've loaded you'll never reload, so refresh etc won't ever work

Reply:
1. We still want to do the work, since the content could have changed.
2. When the saga finishes loading (`item.progress === 'loaded'`), it'll still reload.

Reply:
Right, we file saga in `componentWillMount` so it always fires. We just don't override the subdir entry when a folder loads, if the subdir entry is loaded. But when user navigates into the subdir, saga gets fired again and eventually refreshes the list.

Reply:
I'm gonna address feedbacks in https://github.com/keybase/client/pull/10666 if there's any. Will also replace `.map` with `.filter` here in that PR since that's what it essentially does.